backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/OperationInterpreter.java

```diff
@@ -83,15 +83,20 @@ public ToolInterpreter(AQLInterpreter interpreter, IObjectService objectService,
     }
 
     public IStatus executeTool(Tool tool, VariableManager variableManager) {
-        Optional<VariableManager> result = this.executeOperations(tool.getBody(), variableManager);
+        return this.executeOperations(tool.getBody(), variableManager);
+    }
+
+    public IStatus executeOperations(List<Operation> operations, VariableManager variableManager) {
+        Optional<VariableManager> result = this.doExecuteOperations(operations, variableManager);
```

Comment:
Why create another entry point to this execution process when both seems to use the almost same type of objects?

Reply:
I split into two classes: an OperationInterperter for common operation, and a DiagramOperationInterpreter for Diagram tools.